contracts/pre-propose/cwd-pre-propose-approval-single/README.md

```diff
@@ -0,0 +1,68 @@
+# Single choice proposal approval contract
+
+This contract implements an approval flow for proposals, it also handles deposit logic. It works with the `cwd-proposal-single` proposal module.
+
+## Approval Logic
+
+This contract is instantatied with an `approver` address. This address is allowed to approve or reject the proposal.
+
+```
+      ┌──────────┐
+      │          │
+      │  Account │
+      │          │
+      └─────┬────┘
+            │
+            │ Makes prop
+            ▼
+┌────────────────────────┐               ┌────────────────────────┐
+│                        │               │                        │
+│  Pre-propose Approval  │ ◄─────────────┤    Approver Address    │
+│                        │    Approves   │                        │
+└───────────┬────────────┘    or rejects └────────────────────────┘
+            │
+            │ Creates prop
+            │ on approval
+            ▼
+┌────────────────────────┐
+│                        │
+│     Proposal Single    │
+│                        │
+└───────────┬────────────┘
+            │
+            │ Normal votin
+            │
+            ▼
+┌────────────────────────┐
+│                        │
+│       Main DAO         │
+│                        │
+└────────────────────────┘
+```
+
+The `approver` may also register a `ProposalSubmitHook`, which fires every time a proposal is submitted to the `cwd-pre-propose-approval-single` contract.
+
+## Deposit Logic
+
+It may accept either native ([bank
+module](https://docs.cosmos.network/main/modules/bank/)),
+[cw20](https://github.com/CosmWasm/cw-plus/tree/bc339368b1ee33c97c55a19d4cff983c7708ce36/packages/cw20)
+tokens, or no tokens as a deposit. If a proposal deposit is enabled
+the following refund strategies are avaliable:
+
+1. Never refund deposits. All deposits are sent to the DAO on proposal
+   completion.
+2. Always refund deposits. Deposits are returned to the proposer on
+   proposal completion.
+3. Only refund passed proposals. Deposits are only returned to the
+   proposer if the proposal passes. Otherwise, they are sent to the
+   DAO.
```

Comment:
```suggestion
1. Never refund deposits. All deposits are sent to the DAO on proposal
   completion.
2. Always refund deposits. Deposits are returned to the proposer on
   proposal completion and even rejection by the `approver`.
3. Only refund passed proposals. Deposits are only returned to the
   proposer if the proposal is approved and passes. Otherwise, they
   are sent to the DAO.
```